src/compiler/parser/index.js

```diff
@@ -598,6 +598,15 @@ function makeAttrsMap (attrs: Array<Object>): Object {
     }
     map[attrs[i].name] = attrs[i].value
   }
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    map['v-model'] &&
+    (map['v-bind:value'] || map[':value']) &&
+    (!map['type'] || map['type'] === 'text')
```

Comment:
This warning should apply to anything that is not:
- type=checkbox
- type=radio
- select